lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -62,52 +62,44 @@ ui32 CGHeroInstance::getTileCost(const TerrainTile &dest, const TerrainTile &fro
 	unsigned ret = 100;
```

Comment:
Actually "terrCosts" it's data from "terrains.json", but terrain-specific cost only used when your army is not native for that terrain and this is what this loop checking.